vmngclient/api/template_api.py

```diff
@@ -56,48 +54,141 @@ def __init__(self, name):
         self.message = f"Template: {name} - wrong template type."
 
 
+class DeviceTemplateFeature(Enum):
+    LAWFUL_INTERCEPTION = "lawful-interception"
+    CLOUD_DOCK = "cloud-dock"
+    NETWORK_DESIGN = "network-design"
+    VMANAGE_DEFAULT = "vmanage-default"
+    ALL = "all"
+
+
 class TemplatesAPI:
     def __init__(self, session: vManageSession) -> None:
         self.session = session
 
-    @property
-    def templates(self) -> List[Template]:
-        """
+    def _get_feature_templates(
+        self, summary: bool = True, offset: Optional[int] = None, limit: Optional[int] = None
+    ) -> DataSequence[FeatureTemplateInfo]:
+        """In a multitenant vManage system, this API is only available in the Provider view."""
```

Comment:
Do we want to block it by checking `session_type`? Or at least check if session type is provider and if not log warning.

Reply:
Probably we need to discuss that internally. There are several endpoints working only on provider view and we do nothing about that.